tests/test_ranges.py

```diff
@@ -617,3 +617,135 @@ def test_days_property_has_correct_value(self):
+                        end=datetime.date(2000, 1, 4),
+                    ),
+                ),
+                (
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 3),
+                        end=datetime.date(2000, 1, 4),
+                    ),
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 1),
+                        end=datetime.date(2000, 1, 2),
+                    ),
+                ),
+            ]
+        )
+        def test_handles_adjacent_ranges(self, range, other):
+            """
+            As FiniteDateRange boundaries are double inclusive, when two ranges
+            end/start on consecutive days they effectively cover a fully contiguous
+            period of time and should make a union.
+            """
+            union = range | other
+            assert union == ranges.FiniteDateRange(
+                start=datetime.date(2000, 1, 1),
+                end=datetime.date(2000, 1, 4),
+            )
+
+        def test_handles_ranges_out_of_order(self):
+            """
+            The order of the ranges should not matter when creating a union.
+            """
+            ranges_ = [
+                ranges.FiniteDateRange(
+                    start=datetime.date(2000, 1, 3),
+                    end=datetime.date(2000, 1, 4),
+                ),
+                ranges.FiniteDateRange(
+                    start=datetime.date(2000, 1, 1),
+                    end=datetime.date(2000, 1, 2),
+                ),
+            ]
+            union = ranges_[0] | ranges_[1]
+            assert union == ranges.FiniteDateRange(
+                start=ranges_[1].start,
+                end=ranges_[0].end,
+            )
+
+        @pytest.mark.parametrize(
+            "range, other", [
+                (
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 1),
+                        end=datetime.date(2000, 1, 2),
+                    ),
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 4),
+                        end=datetime.date(2000, 1, 5),
+                    ),
+                ),
+                (
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 4),
+                        end=datetime.date(2000, 1, 5),
+                    ),
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 1),
+                        end=datetime.date(2000, 1, 2),
+                    ),
+                ),
+            ]
+        )
+        def test_handles_disjoint_ranges(self, range, other):
+            """
+            Disjoint ranges (those with a full clear day between them) should not make
+            a union.
+            """
+            union = range | other
+            assert union is None
+
+        @pytest.mark.parametrize(
```

Comment:
:panda_face: this test is complex enough that it should probably have ids on its cases.

Reply:
🆗